upstairs/src/test.rs

```diff
@@ -5105,4 +5245,882 @@ mod up_test {
         // The two writes and the flush should be completed.
         assert_eq!(up.downstairs.lock().await.completed.len(), 3);
     }
+
+    // Job dependency tests
+
+    #[tokio::test]
+    async fn test_deps_writes_depend_on_overlapping_writes() {
+        // Test that the following job dependency graph is made:
+        //
+        // write @ 0 -> write @ 0
+
+        let upstairs = make_upstairs();
+        upstairs.set_active().await.unwrap();
+
+        upstairs
+            .submit_write(
+                Block::new_512(0),
+                Bytes::from(vec![0xff; 512]),
+                None,
+                false,
+            )
+            .await
+            .unwrap();
+
+        upstairs
+            .submit_write(
+                Block::new_512(0),
+                Bytes::from(vec![0x00; 512]),
+                None,
+                false,
+            )
+            .await
+            .unwrap();
+
+        let ds = upstairs.downstairs.lock().await;
+        let keys: Vec<&u64> = ds.active.keys().sorted().collect();
+        let jobs: Vec<&DownstairsIO> =
+            keys.iter().map(|k| ds.active.get(k).unwrap()).collect();
+        assert_eq!(jobs.len(), 2);
+
+        assert!(jobs[0].work.deps().is_empty());
+        assert_eq!(jobs[1].work.deps(), &vec![jobs[0].ds_id]);
+    }
+
+    #[tokio::test]
+    async fn test_deps_writes_depend_on_overlapping_writes_chain() {
+        // Test that the following job dependency graph is made:
+        //
+        //           -> write @ 0 ->
+        // write @ 0                 write @ 0
+        //           -------------->
+        //
```

Comment:
This one I don't understand, like why is the second column the only thing in that row and not like this:

```
        // write @ 0 -> write @ 0 ->  write @ 0
        //           -------------->
```

Reply:
Hopefully 5595ccc addresses this